internal/databases/postgres/daemon_handler.go

```diff
@@ -0,0 +1,176 @@
+package postgres
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type SocketMessageType byte
+
+const (
+	CheckType    SocketMessageType = 'C'
+	FileNameType SocketMessageType = 'F'
+)
+
+type SocketMessageHandler interface {
+	Handle(messageBody []byte) error
+}
+
+type ArchiveMessageHandler struct {
+	messageType SocketMessageType
+	fd          net.Conn
+	uploader    *WalUploader
+}
+
+func (h *CheckMessageHandler) Handle(messageBody []byte) error {
```

Comment:
CheckMessageHandler/ArchiveMessageHandler struct declarations and Handle methods are mixed up, you can reorder them